### ISSUE TITLE
ThingShadow unsubscribe acknowledgments cause TypeError

### ISSUE THREAD

**Issue by mwm-twx** [aws/aws-iot-device-sdk-js#66]
Unsubscribe calls will trigger a TypeError when the acknowledgment arrives:

``` javascript
./node_modules/aws-iot-device-sdk/node_modules/mqtt/lib/store.js:76
cb(null, packet);
   ^

TypeError: cb is not a function
   at Store.del (./node_modules/aws-iot-device-sdk/node_modules/mqtt/lib/store.js:76:5)
   at MqttClient._handleAck (./node_modules/aws-iot-device-sdk/node_modules/mqtt/lib/client.js:836:26)
   at MqttClient._handlePacket (./node_modules/aws-iot-device-sdk/node_modules/mqtt/lib/client.js:276:12)
   at process (./node_modules/aws-iot-device-sdk/node_modules/mqtt/lib/client.js:227:12)
   at Writable.writable._write (./node_modules/aws-iot-device-sdk/node_modules/mqtt/lib/client.js:237:5)
   at doWrite (./node_modules/aws-iot-device-sdk/node_modules/mqtt/node_modules/readable-stream/lib/_stream_writable.js:237:10)
   at writeOrBuffer (./node_modules/aws-iot-device-sdk/node_modules/mqtt/node_modules/readable-stream/lib/_stream_writable.js:227:5)
   at Writable.write (./node_modules/aws-iot-device-sdk/node_modules/mqtt/node_modules/readable-stream/lib/_stream_writable.js:194:11)
   at TLSSocket.ondata (_stream_readable.js:536:20)
   at emitOne (events.js:77:13)
```

This appears to be caused by treating subscribe and unsubscribe on the underlying MQTT connection the same, though they have different signatures.  MQTTClient.unsubscribe does not take an 'options' parameter, so that value is taken as the callback function.  Since there is no explicit check on the type of the callback parameter, a TypeError occurs.  The user-supplied callback function is never invoked.

**Comment by rongsaws** [comment]
Thanks for looking into this and providing the root cause analysis. It does look like an issue with calling the underneath unsubscribe() call with incorrect signature. We'll be getting a fix out for that soon.

**Comment by mwm-twx** [comment]
No problem.  Thanks for the quick reply.

**Comment by sbruton** [comment]
+

**Comment by sbruton** [comment]
https://github.com/aws/aws-iot-device-sdk-js/pull/79

**Comment by liuszeng** [comment]
Addressed in release [v1.0.13](https://github.com/aws/aws-iot-device-sdk-js/releases/tag/v1.0.13).
